app/controllers/dradis/plugins/calculators/cvss/issues_controller.rb

```diff
@@ -22,15 +32,27 @@ def update
     def set_cvss_vector
       # Undefined Temporal and Environmental default to X
       @cvss_vector = Hash.new { |h, k| h[k] = 'X' }
-      field_value  = @issue.fields['CVSSv3.Vector'] || @issue.fields['CVSSv3Vector']
+      @cvss4_vector = Dradis::Plugins::Calculators::CVSS::V4::DEFAULT_CVSS_V4.clone
+      field_value_v3 = @issue.fields['CVSSv3.Vector'] || @issue.fields['CVSSv3Vector']
+      field_value_v4  = @issue.fields['CVSSv4.BaseVector']
 
       # If no vector is set yet, that's OK
-      return if field_value.blank?
+      return if field_value_v3.blank? && field_value_v4.blank?
```

Comment:
This logic is somewhat similar to what you have in #edit which may lend itself to some optimisation.